src/code/z_message.c

```diff
@@ -1303,29 +1303,29 @@ void Message_Decode(PlayState* play) {
             // Substitute the player name control character for the file's player name.
             for (playerNameLen = ARRAY_COUNT(gSaveContext.save.info.playerData.playerName); playerNameLen > 0;
                  playerNameLen--) {
-                if (gSaveContext.save.info.playerData.playerName[playerNameLen - 1] != 0x3E) {
+                if (gSaveContext.save.info.playerData.playerName[playerNameLen - 1] != FILENAME_SPACE) {
                     break;
                 }
             }
             // "Name"
             PRINTF("\n名前 ＝ ");
             for (i = 0; i < playerNameLen; i++) {
                 curChar2 = gSaveContext.save.info.playerData.playerName[i];
-                if (curChar2 == 0x3E) {
+                if (curChar2 == FILENAME_SPACE) {
                     curChar2 = ' ';
-                } else if (curChar2 == 0x40) {
+                } else if (curChar2 == FILENAME_PERIOD) {
                     curChar2 = '.';
-                } else if (curChar2 == 0x3F) {
+                } else if (curChar2 == FILENAME_DASH) {
                     curChar2 = '-';
-                } else if (curChar2 < 0xA) {
-                    curChar2 += 0;
+                } else if (curChar2 <= FILENAME_DIGIT('9')) {
+                    curChar2 -= FILENAME_DIGIT('0');
                     curChar2 += '0';
-                } else if (curChar2 < 0x24) {
-                    curChar2 += 0;
-                    curChar2 += '7';
-                } else if (curChar2 < 0x3E) {
-                    curChar2 += 0;
-                    curChar2 += '=';
+                } else if (curChar2 <= FILENAME_UPPERCASE('Z')) {
```

Comment:
Is this function matching for NTSC? There seems to be a gap between encoded 9 and encoded A
(doesn't need to be matching now just asking)

This is peak programming if originally written like this with macros btw, it looks completely random that the successive range checks work out as having increasing bounds. I wonder if instead of
`if (val < a) { /*0..a*/ } else if (val < b) { /*a:b*/ } ...`
this would match (extra superfluous `val >= a` check)
`if (val < a) { /*0:a*/ } else if (val >= a && val < b) { /*a:b*/ } ...`
I forgot how smart(n't) ido is on this

Reply:
Yeah this part is matching for NTSC, it's a big motivation for this change

Adding the `val >= a` checks didn't seem to work but I didn't try very hard. But the `val >= a` check wouldn't work for NTSC anyway because of the gap between 9 and A right?

Reply:
Alright then
What are the bytes between encoded '9' and encoded 'A' for?

If they're actually used and (I suppose) encoding japanese characters would that mean the "A-Z decoding" branch also decodes those characters?

Reply:
Yeah they're used for Japanese characters. This part in particular translates the filename into English text, so I think it just breaks horribly if you play in English but have a Japanese filename somehow. Eventually there will be a separate code path for Japanese text near here. (Maybe it would have been less confusing to package this change in with the z_message.c changes, sorry)

Reply:
Wait so is this matching NTSC or not?

Reply:
This part does match but Message_Decode as a whole still needs more work. https://github.com/cadmic/oot/tree/gc-us has gc-us OK and here's the rest of the Message_Decode changes (still WIP): https://github.com/cadmic/oot/commit/42af8587fe0597e490fad6c9dc85c0df3b395a64#diff-3259eac13a0c0ac44581d66dc29035e7fcfa5d4fea7ff8451c63cf10ad50de43L1247